src/pyFAI/integrator/fiber.py

```diff
@@ -196,269 +321,196 @@ def integrate_grazing_incidence(self, data,
-            incident_angle = incident_angle or integrated_unit.incident_angle
-        else:
-            incident_angle = incident_angle or 0.0
-
-        if isinstance(integrated_unit, units.UnitFiber):
-            if integrated_unit.incident_angle != incident_angle:
-                integrated_unit.set_incident_angle(incident_angle)
-                logger.info(f"Incident angle set to {incident_angle} for unit {integrated_unit}. AzimuthalIntegrator will be reset.")
-                reset = True
-        else:
-            integrated_unit = units.to_unit(integrated_unit)
-            integrated_unit.set_incident_angle(incident_angle)
-            logger.info(f"Incident angle set to {incident_angle} for unit {integrated_unit}. AzimuthalIntegrator will be reset.")
-            reset = True
-
-        if isinstance(output_unit, units.UnitFiber):
-            if output_unit.incident_angle != incident_angle:
-                output_unit.set_incident_angle(incident_angle)
-                logger.info(f"Incident angle set to {incident_angle} for unit {output_unit}. AzimuthalIntegrator will be reset.")
-                reset = True
-        else:
-            output_unit = units.to_unit(output_unit)
-            output_unit.set_incident_angle(incident_angle)
-            logger.info(f"Incident angle set to {incident_angle} for unit {output_unit}. AzimuthalIntegrator will be reset.")
-            reset = True
-
-        if isinstance(integrated_unit, units.UnitFiber):
-            tilt_angle = tilt_angle or integrated_unit.tilt_angle
-        else:
-            tilt_angle = tilt_angle or 0.0
-
-        if isinstance(integrated_unit, units.UnitFiber):
-            if integrated_unit.tilt_angle != tilt_angle:
-                integrated_unit.set_tilt_angle(tilt_angle)
-                logger.info(f"Tilt angle set to {tilt_angle} for unit {integrated_unit}. AzimuthalIntegrator will be reset.")
-                reset = True
-        else:
-            integrated_unit = units.to_unit(integrated_unit)
-            integrated_unit.set_tilt_angle(tilt_angle)
-            logger.info(f"Tilt angle set to {tilt_angle} for unit {integrated_unit}. AzimuthalIntegrator will be reset.")
-            reset = True
 
-        if isinstance(output_unit, units.UnitFiber):
-            if output_unit.tilt_angle != tilt_angle:
-                output_unit.set_tilt_angle(tilt_angle)
-                logger.info(f"Tilt angle set to {tilt_angle} for unit {output_unit}. AzimuthalIntegrator will be reset.")
-                reset = True
-        else:
-            output_unit = units.to_unit(output_unit)
-            output_unit.set_tilt_angle(tilt_angle)
-            logger.info(f"Tilt angle set to {tilt_angle} for unit {output_unit}. AzimuthalIntegrator will be reset.")
-            reset = True
+        if "npt_output" or "npt_integrated" in kwargs:
+            logger.warning(f"npt_output and npt_integrated are deprecated parameters. Use npt_oop and npt_ip instead")
 
```

Comment:
2 things:
* Create a compatibiliy layer for now
* declare since when this is deprecated and when it will be removed (~2 years).
Have a look at the `pyFAI/utils/decorator` module which contains some helper for deprecation warning.